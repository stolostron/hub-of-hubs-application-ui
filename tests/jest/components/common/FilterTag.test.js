/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright Contributors to the Open Cluster Management project
"use strict";

import React from "react";
import Tag from "../../../../src-web/components/common/FilterTag";
import renderer from "react-test-renderer";

describe("FilterTag component", () => {
  it("renders as expected", () => {
    const props = {
      classNames: {
        selectedTag: "selectedTag1",
        selectedTagName: "selectedTagName1"
      },
      tag: { classType: "FilterTag" },
      name: "TestingFilterTag1",
      onDelete() {}
    };
    const component = renderer.create(<Tag {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders as expected", () => {
    const props = {
      classNames: {
        selectedTag: "selectedTag2",
        selectedTagName: "selectedTagName2"
      },
      tag: { classType: "keyword" },
      name: "TestingFilterTag2",
      onDelete() {}
    };
    const component = renderer.create(<Tag {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
