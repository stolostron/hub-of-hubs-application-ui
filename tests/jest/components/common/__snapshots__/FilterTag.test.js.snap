// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTag component renders as expected 1`] = `
<button
  className="selectedTag1"
  onClick={[Function]}
  type="button"
>
  <span
    className="selectedTagName1"
  />
</button>
`;

exports[`FilterTag component renders as expected 2`] = `
<button
  className="react-tags__keyword-tag"
  onClick={[Function]}
  type="button"
>
  <span
    className="selectedTagName2"
  />
</button>
`;
